drivers/usb/device/usb_dc_mcux.c

```diff
@@ -486,8 +486,6 @@ int usb_dc_ep_write(const uint8_t ep, const uint8_t *const data,
 		    const uint32_t data_len, uint32_t *const ret_bytes)
 {
 	uint8_t ep_abs_idx = EP_ABS_IDX(ep);
-	uint8_t *buffer = (uint8_t *)dev_state.eps[ep_abs_idx].block.data;
```

Comment:
This intermediate buffer is not needed anymore. 
https://github.com/zephyrproject-rtos/zephyr/blob/e7443bb2035574ec1377b791032769e3d0a7f428/drivers/usb/device/usb_dc_mcux.c#L281
Should it still be allocated?

Reply:
This is used for the receive.

Reply:
It is allocated for both IN and OUT endpoints, but with this changes the driver does not need to allocate it for the IN endpoints.